server/resp/src/main/java/org/infinispan/server/resp/commands/cluster/SLOTS.java

```diff
@@ -91,8 +92,9 @@ private static CompletionStage<CharSequence> getSlotsInformation(Resp3Handler ha
                StringBuilder builder = new StringBuilder();
 
                Map<List<Address>, IntSet> segmentOwners = new HashMap<>();
-               for (int i = 0; i < hash.getNumSegments(); i++) {
-                  segmentOwners.computeIfAbsent(hash.locateOwnersForSegment(i), ignore -> IntSets.mutableEmptySet())
+               for (int i = 0; i < SLOT_SIZE; i++) {
+                  int s = i % hash.getNumSegments();
```

Comment:
With the change I suggested below this becomes `int s = i / hash.getNumSegments()`

To be honest you could also optimize and combine this and reverse the loop to  be based on segments instead. Then for the first non contiguous segment you write the smallest slot it maps to ie. `SLOT_SIZE / numSegment`. And then y ou find the next non contiguous segment you write the highest value of the last segment and repeat.

This then requires no map lookup and is not O(16384*numSegments) and is just O(numSegments)